File: source/logging/LogInfo.java
package logging;

import utility.Time;

public class LogInfo {
    public final LogType type;
    public final String firstMessagePart;
    public final String secondMessagePart;
    public final int senderLength;

    public LogInfo(LogType type, String sender, Object object) {
        this.type = type;
        firstMessagePart = type + " " + Time.now() + "  " + sender;
        secondMessagePart = object.toString();
        senderLength = sender.length();
    }
}

File: source/logging/LogInfo.kt
package logging

import utility.Time

class LogInfo(val type: LogType, sender: String, obj: Any) {
    val firstMessagePart: String
    val secondMessagePart: String
    val senderLength: Int

    init {
        firstMessagePart = type.toString() + " " + Time.now() + "  " + sender
        secondMessagePart = obj.toString()
        senderLength = sender.length
    }
}